docs: update, mostly for AppImage notes

GitMensch opened the pull request:
CONTRIBUTING.md:
* add license information
* add conventional commits convention

README.md:
* updated AppImage notes - point to latest/continuous release instead of CI artifacts
* case adjustments for Hotspot and AppImage
* http -> https
* some minor rewording

HACKING.md:
* drop the unzip step as it is not needed any more
* replace shown version string by a clear pseudo-version
* update shown dates to a more current one

lievenhey commented:
we are now using conventional commits. Please prefix your commit message with `chore: `

GitMensch commented:
Wouldn't docs be more appropriate?

GitMensch commented:
Added a second commit for the conventional commits (I've totally missed that) in CONTRIBUTING.md and also updated the HACKING file. Note that Markdownlint is wrong - dropping the double whitespace will lead to the text being all on one line (the formatting rule was always two spaces and a line break means a line break). Should we still pull this or use "one long rendered line" or "adding a paraghraph for each sentence?

milianw commented on HACKING.md:
```diff
@@ -153,22 +153,20 @@ To do that, you will need to download the debuginfo artifact that matches the Ap
 $ tar -xvf hotspot-debuginfo-*.tar.bz2
-hotspot-debuginfo-v1.4.0-43-g02fd82c/usr/bin/hotspot
-hotspot-debuginfo-v1.4.0-43-g02fd82c/usr/lib64/libexec/hotspot-perfparser
+hotspot-debuginfo-v1.5.0-17-g0ed41cf/usr/bin/hotspot
```
here and below. unrelated changes?

GitMensch replied:
We could make it possibly even more obvious: leaving the sha as is, but changing the version to 1.2.3.-45 - Do you agree?
Are you fine with changing the date in the example below or should I leave this as-is?

milianw reviewed (commented):
why the changes to the version + sha? that's just an example

GitMensch commented:
I wanted to use a newer version and date as this makes the document look less outdated to me. If you want to I'll only update the 1.3 one to the 1.4 (to make samples use the same) and don't change the date.

Am 17. Mai 2024 09:44:09 MESZ schrieb Milian Wolff ***@***.***>:
>@milianw commented on this pull request.
>
>why the changes to the version + sha? that's just an example
>
>>  $ tar -xvf hotspot-debuginfo-*.tar.bz2
>-hotspot-debuginfo-v1.4.0-43-g02fd82c/usr/bin/hotspot
>-hotspot-debuginfo-v1.4.0-43-g02fd82c/usr/lib64/libexec/hotspot-perfparser
>+hotspot-debuginfo-v1.5.0-17-g0ed41cf/usr/bin/hotspot
>
>here and below. unrelated changes?
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/KDAB/hotspot/pull/639#pullrequestreview-2062564354
>You are receiving this because you authored the thread.
>
>Message ID: ***@***.***>

lievenhey commented:
@GitMensch can you rebase this patch please. Master contains a patch that should stop `pre-commit` from complaining about that formatting.

GitMensch commented:
Done. To care for @milianw's comments I've also used the previous SHA in the version numbers referenced (less changes) and changed the version number itself top a clear pseudo-version so it won't be likely for any one to feel the need to update them.
